napari/utils/validators.py

```diff
@@ -82,3 +82,20 @@ def func(obj):
                     )
 
     return func
+
+
+def validate_increasing(values: Iterable):
+    """
+    Checks if values in an iterable are increasing and raises an error otherwise.
+    """
+    value = values[0]
```

Comment:
this is going to raise a potentially confusing `IndexError` if it's not at least length 2.  Also, the current error message seems backwards?  (predecessor is the one that came before... so each item should be larger).  lastly, not all iterables are indexable (sequences are, but not iterables)... itertools gives us a nice `tee` function that's handy here for any size iterable:

```python
from itertools import tee

def pairwise(iterable):
    """pairwise(s) == [(s[0], s[1]), (s[1], s[2]), (s[2], s[3]), ...]"""
    a, b = tee(iterable)
    next(b, None)
    return zip(a, b)

def validate_increasing(values: Iterable) -> None:
    """Raise exception if `values` is not monotonically increasing."""
    if any(a >= b for a, b in pairwise(values)):
        raise TypeError(
            trans._(
                "Sequence {sequence} must be monotonically increasing.",
                deferred=True,
                sequence=values,
            )
        )
```

Reply:
I'm not sure about this suggestion. I don't understand code of that kind and would argue it's harder to maintain then:
```
a, b = tee(iterable)
next(b, None)
return zip(a, b)
```

Reply:
also, the docstring is pretty explanatory even if you don't understand the code no?:
```
"""pairwise(s) == [(s[0], s[1]), (s[1], s[2]), (s[2], s[3]), ...]"""
```

Reply:
I'm not against this change. I'm just saying, I won't send a PR with code I think is too complicated. Feel free to adapt, it's your project, your code. ;-)

Reply:
I like Talley's suggestion. In Python 3.10 there is [itertools.pairwise](https://docs.python.org/3/library/itertools.html#itertools.pairwise), so we can eventually drop the `pairwise` definition and just use that directly.

Reply:
Again, I'm not against this. I'm just suggesting to avoid over-engineering to minimize maintenance efforts. At the end, we're talking about checking if `contrast_limits[1]` is larger than `contrast_limits[0]`. Ultimately, this could be done using a single line of code containing some kind of `if`or `assert` statement...

Reply:
it's not over-engineering ... it's doing what the function name says, and doing what your original implementation appeared to be attempting.  If you want to _only_ deal with a list or tuple with exactly two items then:
1. change your type hint to `Sequence` instead of iterable
2. assert len(2)

Reply:
you have to remember that if you add a function called "validate_increasing" to the napari utils, then some future contributor is going to look at it and not have any idea that "robert added this just to check that something with at least two items was strictly increasing".  So, if we're really concerned with minimizing maintenance efforts, we should be strict that the function does what it says it does and never fails...